src/services/reportCache.js

```diff
@@ -108,27 +109,50 @@ export const cacheCourses = async (objectiveId, activityReportObjectiveId, cours
 };
 
 const cacheTopics = async (objectiveId, activityReportObjectiveId, topics = []) => {
-  const topicIds = topics.map((topic) => {
-    if (!topic.id) {
-      auditLogger.error(`Error saving ARO topics: ${JSON.stringify(topics)} for objectiveId: ${objectiveId} and activityReportObjectiveId: ${activityReportObjectiveId}`);
+  // Find all topics with missing ids
+  const topicsNeedingLookup = topics.filter((t) => !t.id && t.name);
```

Comment:
do we know why we might receive topics without an id in the first place? do we only *most of the time* receive topics with an id?